src/main/java/uk/gov/pay/directdebit/common/validation/ApiValidation.java

```diff
@@ -39,9 +75,28 @@ private boolean isFieldSizeValid(Map<String, String> chargeRequest, String field
                 .collect(Collectors.toList());
     }
 
-    private List<String> checkInvalidSizeFields(Map<String, String> inputData, Map<String, Integer> maximumFieldsSize) {
+    private boolean isMinimumFieldSizeValid(Map<String, String> inputData, String fieldName, int fieldSize) {
+        Optional<String> value = Optional.ofNullable(inputData.get(fieldName));
+        return !value.isPresent() || //already checked that mandatory fields are already there
+                value.get().length() >= fieldSize;
+    }
+
+    private List<String> checkInvalidMinimumSizeFields(Map<String, String> inputData, Map<String, Integer> maximumFieldsSize) {
         return maximumFieldsSize.entrySet().stream()
-                .filter(entry -> !isFieldSizeValid(inputData, entry.getKey(), entry.getValue()))
+                .filter(entry -> !isMinimumFieldSizeValid(inputData, entry.getKey(), entry.getValue()))
```

Comment:
I  think having `checkInvalidMinimumSizeFields` and `checkInvalidMaximumSizeFields` is duplicating unnecessarily quite some code. We could expand the size validation to check for both minimum and maximum (ie `value.get().length() < minLength || value.get().length() > maxLength`) instead. 
Instead of `Map<String, Integer> maximumFieldsSize` and `Map<String, Integer> minimumFieldsSize` there are different approaches - a multimap, a proper class with min and max length as fields, a bifunction which takes min and max length and returns a boolean, etc. etc.

 Feel free to discuss this approach if you don't agree :)

Reply:
Thanks @simad , great idea! I've added the class approach